homeassistant/components/autarco/coordinator.py

```diff
@@ -60,8 +62,10 @@ async def _async_update_data(self) -> AutarcoData:
             inverters = await self.client.get_inverters(self.account_site.public_key)
             if site.has_battery:
                 battery = await self.client.get_battery(self.account_site.public_key)
-        except AutarcoConnectionError as error:
-            raise UpdateFailed(error) from error
+        except AutarcoAuthenticationError as err:
+            raise ConfigEntryAuthFailed(err) from err
```

Comment:
How to specific test for R66?

Reply:
Make a test for updating the coordinator data which raises `AutarcoAuthenticationError` and then check there is a reauth flow active.
Typically an entity test and not a config flow test.

Reply:
You can also make it a test in "test_init.py" if it raises on "async_config_entry_first_refresh"

Reply:
Is there some example from other integration that I can use?

Reply:
For init step:
https://github.com/home-assistant/core/blob/3071aa2da1ab20a9df29c135d263006b430f6e82/tests/components/renault/test_init.py#L50-L60

Reply:
I created something like this:
```py
async def test_setup_entry_exception(
    hass: HomeAssistant, mock_autarco_client: AsyncMock, mock_config_entry: MockConfigEntry
) -> None:
    """Test ConfigEntryNotReady when API raises an exception during entry setup."""
    mock_config_entry.add_to_hass(hass)
    mock_autarco_client.get_site.side_effect = AutarcoAuthenticationError

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    assert mock_config_entry.state is ConfigEntryState.SETUP_ERROR
```
But pytest is raising this error message:
"Failed: Translation not found for homeassistant: `issues.config_entry_reauth.title`. Please add to homeassistant/components/homeassistant/strings.json"

Reply:
Make sure that compile the translations before running the test

Reply:
Yes but don't really understand why I need to add this to the strings.json

Reply:
Did you compile all the translations? or only your own translations?
`python3 -m script.translations develop --all`

Reply:
Only for the integration and I also removed my venv last week to start using python 3.13 😅 But now it worked and I get 100% code coverage with pytest.